src/components/videoCard/VideoCard.jsx

```diff
@@ -1,18 +1,29 @@
 import "./VideoCard.css";
 import { useState } from "react";
 import { VideoMenu } from "./VideoMenu";
+import { useNavigate } from "react-router-dom";
+
+export function VideoCard({
+  video,
+  setIsPlaylistModalVisible,
+  setPlaylistModalVideo,
+}) {
 
-export function VideoCard({ video, setIsPlaylistModalVisible, setPlaylistModalVideo }) {
   const [isVideoMenuVisible, setIsVideoMenuVisible] = useState(false);
   const { channelName, videoThumbnail, title, channelThumbnail, views, likes } =
     video;
-
-  const handleVideoMenuToggleClick = () => {
+  const navigate = useNavigate();
+  const handleVideoMenuToggleClick = (e) => {
+    e.stopPropagation();
```

Comment:
`stopPropagation()` is used to stop further propagation of the current event in the capturing and bubbling phases. Here I have used it in `videoMenu` component so that user can toggle it without navigating to the `singleVideoPage` (as on the whole card I have attached an event listener, which onClick navigates user to different page). Basically it stops propagation of the event from child to parent